clang/lib/Analysis/ThreadSafety.cpp

```diff
@@ -2392,6 +2397,8 @@ void ThreadSafetyAnalyzer::runAnalysis(AnalysisDeclContext &AC) {
   for (const auto &Lock : LocksReleased)
     ExpectedFunctionExitSet.removeLock(FactMan, Lock);
 
+  ConstructedObjectMapTy ConstructedObjects;
```

Comment:
I wonder if we should put it into the `ThreadSafetyAnalyzer`, where we already have the similar `LocalVariableMap`. Then we don't need to pass it into `BuildLockset`, because that already has a pointer to the `ThreadSafetyAnalyzer`.